astropy/io/ascii/ui.py

```diff
@@ -220,7 +220,7 @@ def _guess(table, read_kwargs, format, fast_reader):
             reader.guessing = True
             return reader.read(table)
 
-        except (core.InconsistentTableError, ValueError, TypeError,
+        except (core.InconsistentTableError, ValueError, TypeError, AttributeError,
```

Comment:
Not sure if you were implying subclassing `InconsistentTableError` to be even more specific?  That would be an idea as well.